internal/utils/utils.go

```diff
@@ -106,3 +108,17 @@ func (w *WaitObserver) Retry(err error) error {
 
 	return nil
 }
+
+func Exec(log func(format string, a ...interface{}), command string, args ...string) ([]byte, error) {
+	c := exec.Command(command, args...)
+	bytes, err := c.CombinedOutput()
+	if err != nil {
+		log("❌ Unable to execute %s %s:", command, strings.Join(args, " "))
+		if len(bytes) > 0 {
+			log("%s", string(bytes))
+		}
+		return []byte{}, fmt.Errorf("unable to execute \"%s %s\": %w", command, args, err)
```

Comment:
nit: perhaps replace `args` with `strings.Join(args, " ")` to avoid having square brackets in the error: (like `unable to execute "az [ad sp create-for-rbac --output json --name cilium-operator]": exit status 1`)